qsc/write_vmec_input.py

```diff
@@ -0,0 +1,283 @@
+    mpol_nonzero = mpol
+    RBC = np.zeros((int(2*ntor+1),int(mpol+1)))
+    RBS = np.zeros((int(2*ntor+1),int(mpol+1)))
+    ZBC = np.zeros((int(2*ntor+1),int(mpol+1)))
+    ZBS = np.zeros((int(2*ntor+1),int(mpol+1)))
+    factor = 2 / (N_theta * N_phi_conversion)
+    for j_phi in range(N_phi_conversion):
+        for j_theta in range(N_theta):
+            for m in range(mpol):
+                nmin = -ntor
+                if m==0: nmin = 1
+                for n in range(nmin, ntor):
+                    angle = m * theta[j_theta] - n * self.nfp * phi_conversion[j_phi]
+                    sinangle = np.sin(angle)
+                    cosangle = np.cos(angle)
+                    factor2 = factor
+                    # The next 2 lines ensure inverse Fourier transform(Fourier transform) = identity
+                    if np.mod(N_theta,2) == 0 and m  == (N_theta/2): factor2 = factor2 / 2
+                    if np.mod(N_phi_conversion,2) == 0 and abs(n) == (N_phi_conversion/2): factor2 = factor2 / 2
+                    RBC[n,m] = RBC[n,m] + R_2D[j_theta, j_phi] * cosangle * factor2
+                    RBS[n,m] = RBS[n,m] + R_2D[j_theta, j_phi] * sinangle * factor2
+                    ZBC[n,m] = ZBC[n,m] + z_2D[j_theta, j_phi] * cosangle * factor2
+                    ZBS[n,m] = ZBS[n,m] + z_2D[j_theta, j_phi] * sinangle * factor2
+    RBC[0,0] = np.sum(R_2D) / (N_theta * N_phi_conversion)
+    ZBC[0,0] = np.sum(z_2D) / (N_theta * N_phi_conversion)
+
+    if lasym == False:
+        RBS = 0
+        ZBC = 0
+
+    # Write to VMEC file
+    File_object = open(filename,"w+")
+    File_object.write("! This &INDATA namelist was generated by pyQSC\n")
+    if vmec_template_filename!=[]:    File_object.write("! Based on template file ",vmec_template_filename+'\n')
+    File_object.write("! r ="+str(r)+'\n')
+    File_object.write('!----- Runtime Parameters -----\n')
+    File_object.write('&INDATA\n')
+    File_object.write('  DELT = '+str(delt)+'\n')
+    File_object.write('  NITER = '+str(niter)+'\n')
+    File_object.write('  NSTEP = '+str(nstep)+'\n')
+    File_object.write('  TCON0 = '+str(tcon0)+'\n')
+    File_object.write('  NS_ARRAY = '+str(ns_array)[1:-1]+'\n')
+    File_object.write('  FTOL_ARRAY = '+str(ftol_array)[1:-1]+'\n')
+    File_object.write('  NITER_ARRAY = '+str(niter_array)[1:-1]+'\n')
+    File_object.write('  FTOL_ARRAY = '+str(ftol_array)[1:-1]+'\n')
+    File_object.write('!----- Grid Parameters -----\n')
+    File_object.write('  LASYM = '+str(lasym)+'\n')
+    File_object.write('  NFP = '+str(self.nfp)+'\n')
+    File_object.write('  MPOL = '+str(mpol)+'\n')
+    File_object.write('  NTOR = '+str(min(ntor,ntorMax))+'\n')
+    File_object.write('  PHIEDGE = '+str(phiedge)+'\n')
+    File_object.write('!----- Pressure Parameters -----\n')
+    File_object.write('  PRES_SCALE = '+str(pres_scale)+'\n')
+    File_object.write("  PMASS_TYPE = '"+pmass_type+"'\n")
+    File_object.write('  AM = '+str(am)[1:-1]+'\n')
```

Comment:
Don't skip 1st element.

Reply:
The [1:-1] command acts on the string. It simply removes the initial '[' and final ']' characters so that they are not outputted to the VMEC input file.